### PR TITLE
Expose server_header_transformation via Gloo API

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Exposes `server_header_transformation` via the Gloo API

# Context

Currently there is no way to remove the `server` header that is added by envoy, which is insufficient for some organization's security posture.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3769

### CODE REVIEW

File: projects/gloo/api/v1/options/hcm/hcm.proto [hunk 1]
```diff
@@ -24,6 +24,12 @@ message HttpConnectionManagerSettings {
         ALWAYS_FORWARD_ONLY = 4;
     }
 
+    enum ServerHeaderTransformation {
+        OVERWRITE = 0;
+        APPEND_IF_ABSENT = 1;
+        PASS_THROUGH = 2;
+    }
```
Comment (kdorosh) [inline projects/gloo/api/v1/options/hcm/hcm.proto:31]: let's add some comments for our auto-generated docs that explain these settings

File: projects/gloo/api/v1/options/hcm/hcm.proto [hunk 2]
```diff
@@ -66,4 +72,6 @@ message HttpConnectionManagerSettings {
     // For an explanation of these settings see https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#config-core-v3-httpprotocoloptions
     google.protobuf.Duration max_connection_duration = 23 [ (gogoproto.stdduration) = true ];
     google.protobuf.Duration max_stream_duration = 24 [ (gogoproto.stdduration) = true ];
+
+    ServerHeaderTransformation server_header_transformation = 25;
```
Comment (kdorosh) [inline projects/gloo/api/v1/options/hcm/hcm.proto:76]: let's add a comment for docs

File: projects/gloo/pkg/plugins/hcm/plugin_test.go [hunk 3]
```diff
@@ -136,6 +137,7 @@ var _ = Describe("Plugin", func() {
 		Expect(cfg.CommonHttpProtocolOptions.IdleTimeout).To(Equal(gogoutils.DurationStdToProto(hcms.IdleTimeout)))
 		Expect(cfg.CommonHttpProtocolOptions.GetMaxConnectionDuration()).To(Equal(gogoutils.DurationStdToProto(hcms.MaxConnectionDuration)))
 		Expect(cfg.CommonHttpProtocolOptions.GetMaxStreamDuration()).To(Equal(gogoutils.DurationStdToProto(hcms.MaxStreamDuration)))
+		Expect(cfg.GetServerHeaderTransformation()).To(Equal(envoyhttp.HttpConnectionManager_OVERWRITE))
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/hcm/plugin_test.go:140]: let's add a test that confirms that setting another value propagates into the envoy config as expected

**Review by kdorosh — COMMENTED** [review]
very nice, just a couple nits

**Review by mlholland — APPROVED** [review]
LGTM

**Pull request merged**